packages/blade/src/components/Dropdown/DropdownIconButton.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { useDropdown } from './useDropdown';
+import { dropdownComponentIds } from './dropdownComponentIds';
+import { getActionListContainerRole } from '~components/ActionList/getA11yRoles';
+import type { BaseButtonProps } from '~components/Button/BaseButton/BaseButton';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import type { IconButtonProps } from '~components/Button/IconButton';
+import StyledIconButton from '~components/Button/IconButton/StyledIconButton';
+import type { TooltipifyComponentProps } from '~utils/TooltipifyComponent';
+import { TooltipifyComponent } from '~utils/TooltipifyComponent';
+
+type DropdownIconButtonProps = Omit<IconButtonProps, 'onClick'> & {
+  onBlur?: BaseButtonProps['onBlur'];
+  onKeyDown?: BaseButtonProps['onKeyDown'];
+  onClick?: IconButtonProps['onClick'];
+  tooltip?: TooltipifyComponentProps['tooltip'];
+};
+
+const _DropdownIconButton = ({
```

Comment:
I wish Dropdown worked like Menu and didn't require extra wrapper components to be exposed 😞

Reply:
Yes. I feel at some point, we should just expose isSelected on MenuItem and move all these usecases to Menu and keep Dropdown as Select and AutoComplete only.